lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2830,6 +2839,13 @@ static ArrayRef<Decl *> evaluateMembersRequest(
       }
     }
 
+    if (auto *med = dyn_cast<MacroExpansionDecl>(member)) {
+      for (auto node : med->getFlattenedExpansion())
```

Comment:
I don't think we can/should support nested recursive expansion that continues to add declarations in the same scope, because all of the macro-produced names would need to be covered by the first macro declaration anyway. Peer macros have the same problem, and I plan on banning recursive peer macro expansion for this reason. The compiler doesn't have a good way to reason about the covered names.